mage/query-modules/python/gnn-node-classification.md

```diff
@@ -0,0 +1,342 @@
+- `weight_decay (float)`: weight decay
+- `split_ratio (float)`: ratio between training and validation data
+- `metrics (list)`: list of metrics to be calculated
+- `node_id_property (str)`: name of nodes id property
+- `num_epochs (int)`: number of epochs
+- `console_log_freq (int)`: frequency of logging metrics
+- `checkpoint_freq (int)`: frequency of saving models
+- `device_type (str)`: cpu or cuda
+- `path_to_model (str)`: path where model is load and saved
+
+#### Usage:
+
+```cypher
+  CALL node_classification.set_model_parameters(
+    {layer_type: "GATJK", learning_rate: 0.001, hidden_features_size: [16,16], class_name: "fraud", features_name: "embedding"}
+  ) PROCEDURE MEMORY UNLIMITED YIELD * RETURN in_channels;
+```
+
+### train(ctx: mgp.ProcCtx, no_epochs: int = 100)
+
+This function performs training of model. It first declares data, model, optimizer and criterion. Then it performs training.
+#### Input
+
+- `ctx (mgp.ProcCtx)`: context of process
+- `no_epochs (int, optional)`: number of epochs. Defaults to 100.
+
+#### Exceptions
+- `Exception`: raised if graph is empty
+
+#### Return values
+`list` of `mgp.Record` of:
+- `epoch (int)`: epoch number
+- `loss (float)`: loss of model on training data
+- `val_loss (float)`: loss of model on validation data
+- `train_log (list)`: list of metrics on training data
+- `val_log (list)`: list of metrics on validation data
+
+#### Usage
+```cypher
+  CALL node_classification.train(10) YIELD *;
+```
+
+### get_training_data()
+This function is used so user can see logged data from training.
+
+#### Return values
+
+`mgp.Record` of:
+- `epoch (int)`: epoch number of record of logged data row
+- `loss (float)`: loss in logged data row
+- `train_log (mgp.Any)`: training parameters of record of logged data row
+- `val_log (mgp.Any)`: validation parameters of record of logged data row
+
+#### Usage
+```cypher
+  CALL node_classification.get_training_data() YIELD *;
+```
+
+### save_model()
+
+This function saves model to model saving folder. If there are already total of **max_models_to_keep** models in model saving folder, oldest model is deleted.
+
+#### Exception
+- `Exception`: raised if model is not initialized or defined
+
+#### Return values
+`mgp.Record` of
+
+- `path (str)`: path to saved model
+- `status (str)`: status of saving model
+
+#### Usage
+```cypher
+  CALL node_classification.save_model() YIELD *;
+```
+
+### load_model(num: int = 0)
+
+This function loads model from defined folder for saved models.
+
+#### Input
+
+- `num (int, optional)`: ordinary number of model to load from default map. Defaults to 0 (newest model).
+
+#### Return values
+`mgp.Record` of 
+- path (str): path to loaded model
+
+#### Usage
+
+```cypher
+  CALL node_classification.load_model() YIELD *;
+```
+
+### predict(ctx: mgp.ProcCtx, vertex: mgp.Vertex)
+
+This function predicts metrics on one node. It is suggested that user previously
+loads unseen test data to predict on it.
```

Comment:
What does the second sentence refer to?

Reply:
To load the dataset completely before the user starts to test it.

Reply:
I changed wording a bit